packages/sanity/playwright-ct/tests/formBuilder/inputs/PortableText/FocusTrackingStory.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable react/jsx-no-bind */
+import {Path, defineArrayMember, defineField, defineType} from '@sanity/types'
+import React, {useEffect, useState} from 'react'
+import {SanityDocument} from '@sanity/client'
+import {TestWrapper} from '../../utils/TestWrapper'
+import {TestForm} from '../../utils/TestForm'
+
+const SCHEMA_TYPES = [
+  defineType({
+    type: 'document',
+    name: 'test',
+    title: 'Test',
+    fields: [
+      defineField({
+        type: 'array',
+        name: 'body',
+        of: [
+          defineArrayMember({
+            type: 'block',
+            of: [
+              defineArrayMember({
+                type: 'object',
+                name: 'inlineObjectWithTextProperty',
+                fields: [
+                  defineField({
+                    type: 'string',
+                    name: 'text',
+                    components: {
+                      input: (inputProps) => (
+                        <div data-testid="inlineTextInputField">
+                          {inputProps.renderDefault(inputProps)}
+                        </div>
+                      ),
+                    },
+                  }),
+                ],
+              }),
+            ],
+          }),
+          defineArrayMember({
+            type: 'object',
+            name: 'testObjectBlock',
+            fields: [{type: 'string', name: 'text'}],
+            components: {
+              input: (inputProps) => (
+                <div data-testid="objectBlockInputField">
+                  {inputProps.renderDefault(inputProps)}
+                </div>
+              ),
+            },
+          }),
+        ],
+      }),
+    ],
+  }),
+]
+
+export function FocusTrackingStory({
+  focusPath: focusPathFromProps,
+  document,
+}: {
+  focusPath?: Path
+  document?: SanityDocument
+}) {
+  const [focusPath, setFocusPath] = useState<Path>(focusPathFromProps || [])
+  useEffect(() => {
+    ;(window as any).__setFocusPath = (path: Path) => {
+      setFocusPath(path)
+    }
+    return () => {
+      ;(window as any).__setFocusPath = undefined
+    }
+  }, [])
```

Comment:
**question:** is this adding complexity where we may not need it? Here we're adding an intermediate state layer, and the associated specs need to be aware of  `window.__setFocusPath` too.

Since the associated specs are checking the value of `data-focus-path` on the _story_ itself, it feels like we're leaning towards _testing the test_ rather than the underlying implementation.

Reply:
@rob, it's not actually testing the value from there.  Testing the actual focusPath value is out of the scope of these tests, which is rather testing that it's getting the precise cursor location as expected.

The code added here is for the tests to know that the `focusPath` has been propagated to form, in order not to test this condition "too early".